### DISCUSSION TITLE
For building, allow changing output directory from 'dist' to 'outDir' (in tsconfig.json)

### DISCUSSION THREAD

**Discussion by cyraid** [grafana/grafana#44806] [Plugins]
If I change 'outDir' in tsconfig.json it makes no difference and is still bundled in 'dist' folder. I would like for the end result to build elsewhere so it can also work with watch.

Edit: Or maybe even a `grafana-toolkit plugin:build --dest <dir>`

**Comment by cyraid** [comment]
I did manage to add a custom workaround for now in a webpack.config.js:

```
function getWebpackConfig(originalConfig, options) {

  const newConfig = {
    ...originalConfig,
    output: {
      ...originalConfig.output,
      path: "/path/"
    } // output //
  }; // newConfig //

  return newConfig;

} // Function //

module.exports = {
  getWebpackConfig,
};
```

But I mean it could've been a lot simpler. :)

**Reply by mckn** [reply]
We are actually planning on changing approach here and when scaffolding a plugin we will generate a set of configurations for your project but if you want to customize theses or change parts of it you are welcome to do so.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
